sdk/ai/azopenaiassistants/models_custom.go

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package azopenaiassistants
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+)
+
+// OpenAI's reference doc for Assistants: https://platform.openai.com/docs/api-reference/assistants
+
+// MessageAttachmentToolDefinition specifies the tools the files being attached will be
+// associated with.
+type MessageAttachmentToolDefinition struct {
+	// CodeInterpreterToolDefinition should be set if you attach files for the CodeInterpreter tool.
+	CodeInterpreterToolDefinition *CodeInterpreterToolDefinition
+
+	// FileSearchToolDefinition should be set if you attach files for the FileSearchToolDefinition tool.
+	FileSearchToolDefinition *FileSearchToolDefinition
+}
+
+// MarshalJSON implements the json.Marshaller interface for type MessageAttachmentToolDefinition.
+func (m *MessageAttachmentToolDefinition) MarshalJSON() ([]byte, error) {
+	switch {
+	case m.CodeInterpreterToolDefinition != nil:
+		return json.Marshal(m.CodeInterpreterToolDefinition)
+	case m.FileSearchToolDefinition != nil:
+		return json.Marshal(m.FileSearchToolDefinition)
+	default:
+		return nil, errors.New("neither CodeInterpreterToolDefinition or FileSearchToolDefinition were set")
+	}
+}
+
+// UnmarshalJSON implements the json.Unmarshaller interface for type MessageAttachmentToolDefinition.
+func (m *MessageAttachmentToolDefinition) UnmarshalJSON(data []byte) error {
+
+	var v *struct {
```

Comment:
Is this intermediate type really necessary? It looks to be identical to `MessageAttachmentToolDefinition`.

Reply:
I deserialized this incorrectly - they're not sub-objects, it's a union. I'll fix it.

Reply:
Fixed.